packages/graphback-runtime-mongodb/src/OffixMongoDBDataProvider.ts

```diff
@@ -42,13 +75,25 @@ export class OffixMongoDBDataProvider<Type = any, GraphbackContext = any> extend
       }
       // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
       data.version = data.version + 1;
+      if (this.updatedAtField) {
+        data[this.updatedAtField] = new Date();
+      }
       // TODO use findOneAndUpdate to check consistency afterwards
       const result = await this.db.collection(this.collectionName).updateOne({ _id: new ObjectId(idField.value) }, { $set: data });
       if (result.result?.ok) {
-        return data;
+        return this.mapFields(data);
       }
     }
 
     throw new NoDataError(`Cannot update ${this.collectionName}`);
   }
+
+  protected mapFields(document: any): any {
+    document = super.mapFields(document);
+    if (this.createdAtField) {
```

Comment:
This will make the createdAt field get overwritten each time, correct? This should only happen in `create`

Reply:
Actually I am using this function to map fields on docs that the data provider returns. It is very ugly but I will try to explain anyhow. The provider does not actually store a createdAt field in the database. I am using this method to add that field to output documents by getting the timestamp from the document's ObjectId.